### ISSUE TITLE
Logon to HMC failed

### ISSUE THREAD

**Issue by dkeppens** [IBM/ansible-power-hmc#62]
Hi,

I'm using the following playbook :

- hosts: "{{ groups['hmc'][0] }}"
  tasks:
    - name: reboot lpar
      ibm.power_hmc.powervm_lpar_instance:
        hmc_host: "{{ inventory_hostname }}"
        hmc_auth:
          username: "ansusr"
          password: "{{ acompw }}"
        system_name: uxauray01
        vm_name: admin-xt02-lp
        action: restart

"ansusr" is our Ansible user and has been created on the HMCs. I verified and can login with keys (passwordless) or with password from our Ansible controller to all HMCs by command line without issues. Yet the collection (Version 1.4) is returning "Logon to HMC failed" when i run the playbook... What can be causing this ? What am i missing here ? lxml has been installed by pip on the controller and Python version currently is at 3.7.11.
Debug output is :
The full traceback is:
  File "/tmp/ansible_ibm.power_hmc.powervm_lpar_instance_payload_379kz9wn/ansible_ibm.power_hmc.powervm_lpar_instance_payload.zip/ansible_collections/ibm/power_hmc/plugins/modules/powervm_lpar_instance.py", line 982, in poweroff_partition
  File "/tmp/ansible_ibm.power_hmc.powervm_lpar_instance_payload_379kz9wn/ansible_ibm.power_hmc.powervm_lpar_instance_payload.zip/ansible_collections/ibm/power_hmc/plugins/module_utils/hmc_rest_client.py", line 237, in __init__
    self.session = self.logon()
  File "/tmp/ansible_ibm.power_hmc.powervm_lpar_instance_payload_379kz9wn/ansible_ibm.power_hmc.powervm_lpar_instance_payload.zip/ansible_collections/ibm/power_hmc/plugins/module_utils/hmc_rest_client.py", line 251, in logon
    timeout=300)
  File "/tmp/ansible_ibm.power_hmc.powervm_lpar_instance_payload_379kz9wn/ansible_ibm.power_hmc.powervm_lpar_instance_payload.zip/ansible/module_utils/urls.py", line 1390, in open_url
    unredirected_headers=unredirected_headers)
  File "/tmp/ansible_ibm.power_hmc.powervm_lpar_instance_payload_379kz9wn/ansible_ibm.power_hmc.powervm_lpar_instance_payload.zip/ansible/module_utils/urls.py", line 1294, in open
    r = urllib_request.urlopen(*urlopen_args)
  File "/opt/freeware/lib64/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/freeware/lib64/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/opt/freeware/lib64/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/opt/freeware/lib64/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/tmp/ansible_ibm.power_hmc.powervm_lpar_instance_payload_379kz9wn/ansible_ibm.power_hmc.powervm_lpar_instance_payload.zip/ansible/module_utils/urls.py", line 467, in https_open
    return self.do_open(self._build_https_connection, req)
  File "/opt/freeware/lib64/python3.7/urllib/request.py", line 1352, in do_open
    raise URLError(err)
fatal: [uxhmc01c1]: FAILED! => changed=false
  invocation:
    module_args:
      action: restart
      advanced_info: null
      all_resources: null
      delete_vdisks: null
      hmc_auth:
        password: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
        username: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
      hmc_host: uxhmc01c1
      iIPLsource: null
      keylock: null
      max_virtual_slots: null
      mem: null
      npiv_config: null
      os_type: null
      physical_io: null
      proc: null
      proc_unit: null
      prof_name: null
      retain_vios_cfg: null
      state: null
      system_name: uxauray01
      virt_network_config: null
      vm_name: admin-xt02-lp
      volume_config: null
  msg: Logon to HMC failed

**Comment by dkeppens** [comment]
Forgot to mention : in the inventory, ansible_connection has been set to local for the HMC's.

**Comment by dkeppens** [comment]
K, it seems the last update through Ansible Galaxy must've gone wrong somewhere since i finally noticed i still had version 1.3 installed. Therefore, this issue is now obsoleted but has been replaced by a new error :
msg: ResponseNotReady('Idle') is now being returned on every playbook run.
Any and all tips would be greatly appreciated.

**Comment by AnilVijayan** [comment]
The initial error message reported [URLError] mostly likely due to the fact that the HMC host name (uxhmc01c1) is not reachable from the control node. But the current issue may need more logs to debug. Can you run with verbosity 4 (-vvvv) and share the stack trace?

**Comment by dkeppens** [comment]
Hi,

Thanks for looking into this.
As for your request, I could, but, luckily, this can be closed now instead. I just now managed to get it working.
I made two changes :
- I unset http_proxy and https_proxy environment variables for the playbook (Update : After an additional test, this doesn't seem necessary and was not causing the problem)
- I added the 'password:' subentry for the hmc_auth dictionary, even though SSH key authentication has been set up and works fine from the command line. This seems to have done the trick.
Just one last question before i'll close this one, shouldn't passwordless auth work just as well though ?
